result_test.go

```diff
@@ -409,30 +409,42 @@ func TestToMapBy(t *testing.T) {
 
 func TestToSlice(t *testing.T) {
 	tests := []struct {
-		input        []int
-		output       []int
-		want         []int
-		wantedOutCap int
+		input             []int
+		output            []int
+		want              []int
+		wantedOutputCap   int
+		outputIsANewSlice bool
 	}{
 		//output is nil slice
-		{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, []int{}, []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 16},
+		{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, []int{}, []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 16, true},
 		//cap(out)>len(result): we get the same slice, just resized.
-		{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, make([]int, 0, 11), []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 11},
+		{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, make([]int, 0, 11), []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 11, false},
 		//cap(out)==len(result): we get the same slice, just resized.
-		{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, make([]int, 0, 10), []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 10},
-		//cap(out)<len(result): we get a new slice with len(out)=len(result) and cap(out)==16
-		{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, make([]int, 0, 5), []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 10},
+		{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, make([]int, 0, 10), []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 10, false},
+		//cap(out)<len(result): we get a new slice with len(out)=len(result) and cap(out)==10
+		{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, make([]int, 0, 5), []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 10, true},
 		//len(out) > len(result): we get a same slice with len(out)=len(result) and cap(out)==12
-		{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12}, []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 12},
+		{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12}, []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 12, false},
 	}
 
 	for _, test := range tests {
+		initialOutputValue := reflect.ValueOf(test.output)
 		From(test.input).ToSlice(&test.output)
+		modifiedOutputValue := reflect.ValueOf(test.output)
 		if !reflect.DeepEqual(test.output, test.want) {
 			t.Errorf("From(%v).ToSlice()=%v expected %v", test.input, test.output, test.want)
 		}
-		if cap(test.output) != test.wantedOutCap {
-			t.Errorf("cap(output)=%d expected %d", cap(test.output), test.wantedOutCap)
+		if cap(test.output) != test.wantedOutputCap {
+			t.Errorf("cap(output)=%d expected %d", cap(test.output), test.wantedOutputCap)
+		}
+		isNewSlice := initialOutputValue.Pointer() != modifiedOutputValue.Pointer()
```

Comment:
you can just do this to get uintptr: `(*reflect.SliceHeader)(unsafe.Pointer(&v)).Data`